pkg/golinters/nakedret.go

```diff
@@ -72,63 +73,68 @@ type nakedretVisitor struct {
 	maxLength int
 	f         *token.FileSet
 	issues    []result.Issue
-}
-
-func (v *nakedretVisitor) processFuncDecl(funcDecl *ast.FuncDecl) {
-	file := v.f.File(funcDecl.Pos())
-	functionLineLength := file.Position(funcDecl.End()).Line - file.Position(funcDecl.Pos()).Line
+	root      *nakedretVisitor
 
-	// Scan the body for usage of the named returns
-	for _, stmt := range funcDecl.Body.List {
-		s, ok := stmt.(*ast.ReturnStmt)
-		if !ok {
-			continue
-		}
-
-		if len(s.Results) != 0 {
-			continue
-		}
+	// Details of the function we're currently dealing with
+	funcName    string
+	funcLength  int
+	reportNaked bool
+}
 
-		file := v.f.File(s.Pos())
-		if file == nil || functionLineLength <= v.maxLength {
-			continue
-		}
-		if funcDecl.Name == nil {
-			continue
+func hasNamedReturns(funcType *ast.FuncType) bool {
+	if funcType == nil || funcType.Results == nil {
+		return false
+	}
+	for _, field := range funcType.Results.List {
+		for _, ident := range field.Names {
+			if ident != nil {
+				return true
+			}
 		}
-
-		v.issues = append(v.issues, result.Issue{
-			FromLinter: nakedretName,
-			Text: fmt.Sprintf("naked return in func `%s` with %d lines of code",
-				funcDecl.Name.Name, functionLineLength),
-			Pos: v.f.Position(s.Pos()),
-		})
 	}
+	return false
 }
 
 func (v *nakedretVisitor) Visit(node ast.Node) ast.Visitor {
-	funcDecl, ok := node.(*ast.FuncDecl)
-	if !ok {
-		return v
-	}
-
-	var namedReturns []*ast.Ident
-
-	// We've found a function
-	if funcDecl.Type != nil && funcDecl.Type.Results != nil {
-		for _, field := range funcDecl.Type.Results.List {
-			for _, ident := range field.Names {
-				if ident != nil {
-					namedReturns = append(namedReturns, ident)
-				}
-			}
+	var (
+		funcType *ast.FuncType
+		funcName string
+	)
+	switch s := node.(type) {
+	case *ast.FuncDecl:
```

Comment:
`Visit()` from upstream
https://github.com/alexkohler/nakedret/blob/d95c37512/nakedret.go#L196-L235
only updated with the issue reporting.